### ISSUE TITLE
Analyzer to notify if ConfigureFunctionsWorkerDefaults() is accidently deleted while function app development 

### ISSUE THREAD

**Issue by surgupta-msft** [Azure/azure-functions-dotnet-worker#1887]
The app won't work if  ConfigureFunctionsWorkerDefaults() is deleted from Program.cs

Refer **https://github.com/Azure/azure-functions-dotnet-worker/issues/532#issuecomment-1057089141**

**Comment by kshyju** [comment]
I feel this is a user error. Why do you expect the car to run if no engine is present?

FWIW, aspnet does not have any analyzer complaining if you comment out the methods in main. 

```csharp
public static void Main(string[] args)
{
    var builder = WebApplication.CreateBuilder(args);

    // Add services to the container.
    // builder.Services.AddControllers();

    var app = builder.Build();

    // Configure the HTTP request pipeline.
    // app.UseAuthorization();
    // app.MapControllers();

    app.Run();
}
```

**Comment by jviau** [comment]
Yeah, I agree with @kshyju here. I think we can categorize this as expected knowledge a customer will have - if you want a function app to run, you need to call the bootstrap method for it to run.

I also think this analyzer would be more annoying than anything - what if a customer wants to move the `ConfigureFunctionsWorkerDefaults` to be in a different method and file? But it is still ultimately called at startup?

**Comment by davidpetric** [comment]
> Yeah, I agree with @kshyju here. I think we can categorize this as expected knowledge a customer needs to have - if you want a function app to run, you need to call the bootstrap method for it to run.
> 
> I also think this analyzer would be more annoying than anything - what if a customer wants to move the `ConfigureFunctionsWorkerDefaults` to be in a different method and file? But it is still ultimately called at startup?

If this method is vital for the Functions to be working it would make sense that if I have a AzureFunction project and I somehow delete it, it should somehow tell me I did something wrong.

I had this exact issue multiple times(because of git merges I think) and spent few good hours trying to figure out what was wrong.

**Comment by jviau** [comment]
@davidpetric the issue is that there is no way for our analyzer to know that it is the user intent for this to be a function app or not. All we know is that the analyzer was installed - and so what? This is not a guaranteed signal the user intends this to be a function app. They could be just wanting access to some APIs, or some specific piece of our packages. Writing analyzers off `IHostBuilder` for the _absence_ of something is not really feasible, I see this as proving a negative.

There are a plethora of packages and analyzers out there you can install that will do nothing unless you call the right method to bootstrap them at runtime. Functions is no different. We do nothing unless you bootstrap us via `ConfigureFunctionsWorkerDefaults`. If that call is missing, we can only assume that was intentional to remove it. At some point there must be an explicit user gesture to say "this is a function app", and that is the call to `ConfigureFunctionsWorkerDefaults`.
